sql/log_event.cc

```diff
@@ -1932,7 +1934,7 @@ Query_log_event::begin_event(String *packet, ulong ev_offset,
   int2store(p + FLAGS_OFFSET, flags);
 
   p[EVENT_TYPE_OFFSET]= QUERY_EVENT;
```

Comment:
Is `LOG_EVENT_OFFSET` a duplicate of `EVENT_TYPE_OFFSET`?

Reply:
I think it is actually meant to refer to the [binlog magic header](https://mariadb.com/kb/en/1-binlog-events/#log-file-structure), though it seems that has somewhere along the way been confused with `EVENT_TYPE_OFFSET`. I wonder if `LOG_EVENT_OFFSET` should be removed, and its usages either changed to (depending on the usage) `EVENT_TYPE_OFFSET` (for extracting event type)  or `BIN_LOG_HEADER_SIZE` (for usage in `Rotate_log_event` construction).

@andrelkin , @knielsen any thoughts?

Reply:
```console
$ git status
On branch main
Your branch is up to date with 'origin/main'.

nothing to commit, working tree clean
$ git grep -n LOG_EVENT_OFFSET
sql/log.cc:5904:    Rotate_log_event r(new_name + dirname_length(new_name), 0, LOG_EVENT_OFFSET,
sql/log_event.h:95:#define LOG_EVENT_OFFSET 4
sql/sql_repl.cc:2272:  DBUG_PRINT("info", ("log event code %d", (*packet)[LOG_EVENT_OFFSET+1] ));
sql/sql_repl.cc:2539:  event_type= (Log_event_type)((uchar)(*packet)[LOG_EVENT_OFFSET+ev_offset]);
sql/sql_repl.cc:2674:  event_type= (Log_event_type)((uchar)(*packet)[LOG_EVENT_OFFSET + ev_offset]);
sql/sql_repl.cc:2943:        (Log_event_type)((uchar)(*packet)[LOG_EVENT_OFFSET+ev_offset]);
```

Reply:
> LOG_EVENT_OFFSET should be removed ..?

Right.